### ISSUE TITLE
[iOS][Intermittent] VPN status ‘Not connected’ in device settings after being turned on from client

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#8443]
**Important details:**

* The issue is resolved after connecting/disconnecting the VPN from device settings.

**Affected versions:**

* 2.19.0 (2.202310312207);

**Tested Platforms:**

* iOS;

**Prerequisites:**

* Have Mozilla VPN freshly installed and open;

* Have an account with an active subscription for Mozilla VPN;

* Be signed in;

**Steps to reproduce:**

1. Turn on the VPN while allowing the VPN connection.
1. Go to device settings and observe the status of Mozilla VPN.

**Expected result:**

* Mozilla VPN status in device settings is Connected.

**Actual result:**

* Mozilla VPN status in device settings is Not Connected.

**Notes:**

* See attached video;

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-5794)
┆Reporter: Raluca Monica Ilban

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

This doesn’t affect security as the VPN is indeed connected, no? 
It does undermine confidence in the product though so it’s not purely aesthetic.

**Comment by data-sync-user** [comment]
➤ Raluca Monica Ilban commented:

Hi Santiago Andrigo 

Yes, the VPN is connected, only the status is incorrectly shown in device settings. After closing and reopening the Settings screen, or connecting/disconnecting the VPN from device settings, the status is displayed correctly.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Matt Cleinman 

This is currently happening on iOS 16.6 (not on iOS 15 or iOS 17), while using Mozilla VPN 2.22 and performing reverse actions than the ones described in this ticket.

Steps:

  * Fresh install - prod server;
  * Allow VPN connection;

Results:

  1. The VPN is connected while checking device settings;
  2. In the client the VPN is not automatically turned ON → trying to turn it on => server unavailable modal displayed;
  3. Disconnecting from device settings and turn it on from the client - works;

Important note - not reproducing if the post-on-boarding feature is off.

Attaching logs.

[^mozillavpn.log]

Please let us know if this should be tracked separately.

**Comment by data-sync-user** [comment]
➤ Matt Cleinman commented:

I’m not able to reproduce this on iOS 16.3.1. Is it intermittent, or does it happen every time? And what do you mean by “post-on-boarding feature”? Thanks!

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

Matt Cleinman Yes, it is intermittent on my device. We have other 2 devices with different OS versions where it is not reproducing at all.

By post-onboarding-feature I mean the 4 screens before activation screen.

!image-20240517-062313.png|width=418,height=766,alt="image-20240517-062313.png"!

**Comment by data-sync-user** [comment]
➤ Matt Cleinman commented:

Trying on 2.22 downloaaded from App Store on iOS 16.3.1. I’m not able to repro this even after trying 4 times, though I’m trying on stage. 

Some questions to try to figure out if we have a different setup:

  * Are you able to repro on stage, or is it only happening on prod for you?
  * Fresh install after fully removing the app (not doing “Reset and Quit”), right?
  * These are accounts that already have subscriptions, right? Or are you doing in app purchase as part of it?
  * What are you doing for data and privacy screens of onboarding? Leaving them all to the default, or changing them?
  * Are you allowing the notifications permission on initial launch?

Thanks!

**Comment by data-sync-user** [comment]
➤ Matt Cleinman commented:

Thanks for that info. 

You’re only seeing this on iOS 16 currently, right? Out of curiosity, what device are you using? And what language is it set to?

I tried replicating this a bunch of times on iOS 16, and didn’t get the problem even once. I’m trying to figure out what the differences may be.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

I’ve tried today, while using latest TestFlight build - 3865 - after performing a Reset VPN action and a Stage/Prod log in and I did not reproduce it.

It takes longer until activates, but works in the end.

I’ve also tried few times after a fresh install (deleting the app before), and no luck.

Tried 7 times. 

Also, I’ve tried to repro the initial issue, and was not able.

I have and iPad 10, with iOS 16.6.1, and the language is en-US;

Will mark this as Done, and log a separate ticket if the case in the future.
